### PR TITLE
Deprecate Python 3.6 support

### PR DESCRIPTION
**Pull request by Nayuken**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-python-agent/blob/main/CONTRIBUTING.rst) and [code of conduct](https://github.com/newrelic/.github/blob/master/CODE_OF_CONDUCT.md)._

# Overview
Deprecate and remove all testing and support for Python 3.6.

# Related Github Issue
Include a link to the related GitHub issue, if applicable

# Testing
The agent includes a suite of tests which should be used to
verify your changes don't break existing functionality. These tests will run with
Github Actions when a pull request is made. More details on running the tests locally can be found
[here](https://github.com/newrelic/newrelic-python-agent/blob/main/CONTRIBUTING.rst#testing-guidelines),
For most contributions it is strongly recommended to add additional tests which
exercise your changes.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -42,106 +42,104 @@
...
 envlist =
-    python-adapter_cheroot-{py27,py36,py37,py38,py39,py310},
-    python-adapter_gevent-{py27,py36,py37,py38,py310},
-    python-adapter_gunicorn-{py36}-aiohttp1-gunicorn{19,latest},
-    python-adapter_gunicorn-{py36,py37,py38,py39,py310}-aiohttp3-gunicornlatest,
-    python-adapter_uvicorn-{py36,py37}-uvicorn03,
+    python-adapter_cheroot-{py27,py37,py38,py39,py310},
+    python-adapter_gevent-{py27,py37,py38,py310},
+    python-adapter_gunicorn-{py37,py38,py39,py310}-aiohttp3-gunicornlatest,
+    python-adapter_uvicorn-{py37}-uvicorn03,
```
Comment (TimPansino) [inline tox.ini:48]:
```suggestion
    python-adapter_uvicorn-py37-uvicorn03,
```

The braces are for matrixing, where multiple versions can be specified and they're substituted in a matrix. When there's only 1 item left in a group they become unnecessary.

File: tox.ini [hunk 2]
```diff
@@ -42,106 +42,104 @@
...
+    python-framework_graphql-{py27,py37,py38,py39,py310,pypy,pypy36}-graphql02,
+    python-framework_graphql-{py37,py38,py39,py310,pypy36}-graphql03,
     python-framework_graphql-py37-graphql{0202,0203,0300,0301,0302,master},
-    grpc-framework_grpc-{py27,py36}-grpc0125,
-    grpc-framework_grpc-{py36,py37,py38,py39,py310}-grpclatest,
+    grpc-framework_grpc-{py27}-grpc0125,
+    grpc-framework_grpc-{py37,py38,py39,py310}-grpclatest,
     python-framework_pyramid-{pypy,py27,py38}-Pyramid0104,
-    python-framework_pyramid-{pypy,py27,pypy36,py36,py37,py38,py39,py310}-Pyramid0110-cornice,
+    python-framework_pyramid-{pypy,py27,pypy37,py38,py39,py310}-Pyramid0110-cornice,
```
Comment (TimPansino) [inline tox.ini:130]: Is this intentionally upgraded to pypy 3.7?

File: tox.ini [hunk 3]
```diff
@@ -152,7 +150,7 @@ usefixtures =
 [testenv]
 deps =
     # Base Dependencies
-    {py36,py37,py38,py39,py310,pypy36}: pytest==6.2.5
+    {py37,py38,py39,py310,pypy36}: pytest==6.2.5
```
Comment (TimPansino) [inline tox.ini:153]:
```suggestion
    {py37,py38,py39,py310,pypy36,pypy37}: pytest==6.2.5
```

We must have missed this in a different PR, let's change it here.

File: tox.ini [hunk 4]
```diff
@@ -42,106 +42,104 @@
...
-    python-adapter_cheroot-{py27,py36,py37,py38,py39,py310},
-    python-adapter_gevent-{py27,py36,py37,py38,py310},
-    python-adapter_gunicorn-{py36}-aiohttp1-gunicorn{19,latest},
-    python-adapter_gunicorn-{py36,py37,py38,py39,py310}-aiohttp3-gunicornlatest,
-    python-adapter_uvicorn-{py36,py37}-uvicorn03,
+    python-adapter_cheroot-{py27,py37,py38,py39,py310},
+    python-adapter_gevent-{py27,py37,py38,py310},
+    python-adapter_gunicorn-{py37,py38,py39,py310}-aiohttp3-gunicornlatest,
+    python-adapter_uvicorn-{py37}-uvicorn03,
     ; Temporarily testing py36 on the uvicorn version preceeding v0.15
```
Comment (TimPansino) [inline tox.ini:49]:
```suggestion
```

If this is gone the comment is no longer needed.

**Comment by TimPansino** [comment]
Tests for PyPy 3.6 will all fail until we merge an update to the tests to upgrade all PyPy 3.6 tests to 3.7+. We'll take care of that in a separate PR and this one will have to stay blocked.

File: tox.ini (hunk 2 above)
Comment (Nayuken) [reply tox.ini:130]: From the most recent version of main this was added "temporarily disabling pypy36 on pyramid master" as a comment regarding that change so I believe it is.

**Pull request closed without merging**